src/settings.js

```diff
@@ -1232,10 +1232,17 @@ var IN_TEST_HARNESS = 0;
 // If true, enables support for pthreads.
 var USE_PTHREADS = 0;
 
-// Specifies the number of web workers that are preallocated before runtime is
-// initialized. If 0, workers are created on demand.
+// Specifies the number of web workers that are preallocated with the
+// webassembly module before runtime is initialized. If 0, workers are
+// created on demand.
 var PTHREAD_POOL_SIZE = 0;
 
+// Specifies the number of web workers that are preallocated before the
+// webassembly module is compiled. If 0, workers are created on demand. You
+// likely want this to be >= PTHREAD_POOL_SIZE. Note that this starts the
+// webworkers in a state where they have yet to load the webassembly module.
+var PREWARM_PTHREAD_POOL_WORKERS_SIZE = 0
```

Comment:
I think we can remove `_WORKERS` from the name, to better match the other one.

I don't have a good idea about what's best for the prefix, though. The current `PREWARM_` is reasonable, but also `PREALLOCATED_` is the name used in the JS. If we prefer the former, we should rename the JS I think.

Reply:
Are there really two distinct use cases here?  Why not just extend the meaning of the existing PTHREAD_POOL_SIZE option?

Reply:
Yes.  Image the following timeline

Document load - (a) - Wasm Compile - (b) - Post wasm compile - (c) - wasm main running

PTHREAD_POOL_SIZE adds a step at (c) - wherein we guarantee that N threads are allocated and immediately usable by wasm main running. This necessarily waits till the workers are loaded. Note that because this both creates the workers and waits, this waits for the entire worker creation and load for all threads created.

I'm proposing adding a step at (a) that creates the workers. These are not usable yet - the load hasn't been called. This allows the client to choose to either create the thread pool early at (c) using the current PTHREAD_POOL_SIZE option or to elide doing so until their runtime has started.

Reply:
I can't really think of a reason that a client would want to use `PTHREAD_POOL_SIZE` and not use `PREWARM_PTHREAD_POOL_WORKERS_SIZE`.

Reply:
I agree. The naive change would be to ensure PREWARM_PTHREAD_POOL_WORKERS_SIZE is max(PREWARM_PTHREAD_POOL_WORKERS_SIZE , PTHREAD_POOL_SIZE).

In general, though, I don't like behaviors set by settings to be dependent on multiple settings (in this case both PREWARM_PTHREAD_POOL_WORKERS_SIZE and PTHREAD_POOL_SIZE).

Is there a place we could print a warning? I'm also happy to make a change to library_pthread to spawn the max of those two settings if either is set.

Reply:
I think like @sbc100 that perhaps we can extend the current option to just do the new useful functionality in this PR - I'm not sure I understand the responses since, maybe I'm misunderstanding them? What is the downside to not adding a new option?